spec/models/database_spec.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe "Database" do
+  let(:models_todo) {
+    ["Spree::CreditCard", "Spree::Adjustment",
+     "StripeAccount", "ColumnPreference",
+     "Spree::LineItem", "Spree::ShippingMethod",
+     "Spree::ShippingRate"].freeze
+  }
+
+  it "should have foreign keys for models with a belongs_to relationship" do
+    Rails.application.eager_load!
+    model_classes = filter_model_classes
+
+    migrations = generate_migrations(model_classes)
+
+    expect(migrations.length).to eq(0)
+  end
+
+  def filter_model_classes
+    Dir.glob(Rails.root.join('app/models/**/*.rb').to_s)
+      .map do |file|
+        relative_path = Pathname.new(file).relative_path_from(Rails.root.join('app/models')).to_s
+        subdirectory = File.dirname(relative_path)
+        base_name = File.basename(file, '.rb').camelize
+        subdirectory == "." ? base_name : "#{subdirectory.camelize}::#{base_name}"
+      end
+  end
+
+  def generate_migrations(model_classes)
+    migrations = []
+    filter = lambda { |model| models_todo.include?(model) }
+    pending_models = model_classes.select(&filter)
+    model_classes.reject!(&filter)
```

Comment:
FYI Rails has a handy method called `partition` which can put this on one line: 
http://bicyclethief.github.io/cheatsheets/ruby-enumerables.html (I refer to this page _all_ the time!)

No need to change it now though.